### PR TITLE
Add test for SSD, parameterize test for MMR.

### PR DESCRIPTION
**Pull request by JyChang012**

### CODE REVIEW

**Comment by JyChang012** [comment]
> Yes, from my VSCode's mypy checker:
> 
> ```
> Signature of "rerank" incompatible with supertype "BaseReranker"
> ```

pre-commit also runs mypy. Why doesn't the error show? @yinsn

File: src/rsdiv/diversity/ssd.py [hunk 1]
```diff
@@ -9,17 +9,21 @@
...
+        similarity_scores: Optional[np.ndarray] = None,
         embeddings: np.ndarray,
+        inplace: bool = False
     ) -> Sequence[int]:
         assert k > 0, "k must be larger than 0!"
+        if not inplace:
+            embeddings = embeddings.copy()
         selection = np.argmax(quality_scores).item()
         ret = [selection]
         volume = self.gamma * norm(embeddings[selection])
```
Comment (yinsn) [inline src/rsdiv/diversity/ssd.py:29]: two RuntimeWarnings need to be deal with:
```
  /usr/local/lib/python3.9/site-packages/rsdiv/diversity/ssd.py:32: RuntimeWarning: invalid value encountered in divide
    selected_emb /= norm(selected_emb)

```

```
  /usr/local/lib/python3.9/site-packages/rsdiv/diversity/ssd.py:35: RuntimeWarning: invalid value encountered in multiply
    norms *= volume
```

**Review by yinsn — CHANGES_REQUESTED** [review]
minor fixes.

**Review by yinsn — APPROVED** [review]
LGTM!

**Pull request merged**